src/js/app.js

```diff
@@ -9,7 +9,35 @@
    */
   ns.app = {
 
-    init : function () {
+    init: function () {
+      /**
+        * i18n which stands for internalization is being applied to Piskel to allow for all content in Piskel
+        * to be translated
+        *
+        * Currently there is a directory called i18n which contains the locale directory. In this directory, you
+        * will find all the string translations in JSON format. Each file is named as follows: languageCode_countryCode
+        * For example: en_us.json where en is English and us is United States
+        *
+        * The strings that we will be used depend on the window.locale. window.locale is the 4 letter locale
+        * code defined by users of the Piskel library so they can optionally load non-English strings
+        * into the Piskel UI. For example "en_us" or "es_es"
+        * window.locales will contains all strings available, then we will use window.locale to select the language
+        * we want strings to be in. All of this is happening here: var i18n = window.locales[window.locale];
+        * In the case that window.locale is undefined, we set the strings to english by default
+        *
+        * Example Usage:
+        *  In tools/drawing/SimplePen.js
+        * this.helpText = i18n.simplePenDrawingTool()
+        * Now when the user hovers over the Pen tool, they will see the translation based on the locale
+        *
+        * To see how each key becomes a function look at the tasks/build-i18n.js file where we use the MessageFormat API
+      */
+      var i18n = window.locales[window.locale];
```

Comment:
I should have commented this before, but since `window` is a global Javascript object, we should add some namespacing to whatever we add to it in order to protect it against collisions with other JavaScript libraries. For example, imagine if a user was using Piskel and a different library like VideoJS. Perhaps the VideoJS library also uses `window.locale` or `window.locales` but puts something completely different there, like `window.locale = 'English'` or something. We should try to keep our code from colliding with other code by namespacing it. A simple way to add namespacing is prepending our project name in the key. For example `window.piskel_locale` and `window.piskel_locales`.

Note that making this change will require updating the documentation as well as the code-dot-org code. But you can do the code-dot-org change in a separate PR.

Reply:
This is great. I had no idea. Thanks for pointing that out